include/highfive/H5Easy.hpp

```diff
@@ -91,6 +91,24 @@ inline DataSet dump(File& file,
                     DumpMode mode = DumpMode::Create);
 #endif
 
+///
+/// \brief Write an Eigen::Ref object to a new DataSet in an open HDF5 file.
+///
+/// \param file Writeable opened file
+/// \param path Path of the DataSet
+/// \param data eigen matrix to write
+/// \param mode Write mode
+///
+/// \return the newly created DataSet
+///
+#ifdef H5_USE_EIGEN
+template <class T>
+inline DataSet dump(File& file,
```

Comment:
I *think* using `template DerivedT` inline DataSet dump(..., Eigen::DenseBase<DerivedT>){...}` the dump could be resolved for many more Eigen types (Arrays, Matrices, References, expression...). Maybe beyond the scope of the this PR, though.

Reply:
In principle you are right. However, what complicates matters here is that there is also a 
```cpp
template <class T>
inline DataSet dump(File& file,
                    const std::string& path,
                    const T& data,
                    DumpMode mode = DumpMode::Create);
```
that takes all HighFive's know types (`int`, `double`, ..., `std::vector<T>`). The trick is to convince the compiler not to take it when calling with an Eigen object. 

My approach is to include overloads of the actual objects that call `dump` (and not their parents). What could be more elegant is to have a static assert in the overload I mentioned. But I'm not entirely sure how to play that out.

Reply:
Thank you @eudoxos for your input. We will definitely improve on our Eigen support and will take this into consideration.

Reply:
> The trick is to convince the compiler not to take it when calling with an Eigen object.

Even though this is closed, just a note for the future: `std::is_base_of` might help here, perhaps.

Reply:
That would be a good idea @eudoxos ! Evidently I did not search long enough. Would you be willing to give it a try and open a PR?